internal/guest/runtime/hcsv2/spec.go

```diff
@@ -152,3 +167,67 @@ func getGroup(spec *oci.Spec, filter func(user.Group) bool) (user.Group, error)
 	}
 	return groups[0], nil
 }
+
+// applyAnnotationsToSpec modifies the spec based on additional information from annotations
+func applyAnnotationsToSpec(ctx context.Context, spec *oci.Spec) error {
+	// Check if we need to override container's /dev/shm
+	if val, ok := spec.Annotations["io.microsoft.container.storage.shm.size-kb"]; ok {
+		sz, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return errors.Wrap(err, "/dev/shm size must be a valid integer")
+		}
+		if sz <= 0 {
+			return errors.Errorf("/dev/shm size must be a positive integer, got: %d", sz)
+		}
+
+		size := fmt.Sprintf("size=%dk", sz)
+		mt := oci.Mount{
+			Destination: "/dev/shm",
+			Type:        "tmpfs",
+			Source:      "shm",
+			Options:     []string{"nosuid", "noexec", "nodev", "mode=1777", size},
```

Comment:
I was unable to find the exact options used for the default `/dev/shm` mount. Are those options exactly same as these options?

Reply:
yup, I'll put a link https://github.com/jterry75/cri/blob/windows_port/vendor/github.com/containerd/containerd/oci/spec.go#L183-L187